ste/sourceInfoProvider.go

```diff
@@ -74,10 +76,13 @@ type IBlobSourceInfoProvider interface {
 	BlobType() azblob.BlobType
 }
 
-type ISDDLBearingSourceInfoProvider interface {
+type ISMBPropertyBearingSourceInfoProvider interface {
 	ISourceInfoProvider
 
 	GetSDDL() (string, error)
+	GetFileSMBCreationTime() (time.Time, error)
```

Comment:
Question/suggestion: what do you think about

(a) Modifying azFiles.SMBPropertyHolder to drop the FilePermissionKey() that it currently has. Nothing seems to be using that, as far as I can tell. That leaves an interface with just creationTime, lastWriteTime and attributes.
(b) then modifiy ISMBPropertyBearingSourceInfoProvider, so that it has a GetSMBProperties() method, which returns a SMBPropertyHolder.

This would align the AzCopy code with the interface that's already in the files SDK, and I think it would also make some of the code simpler here with less duplication. E.g. in the local source info provider, we'd hit the filesystem just once to ask for the properties, instead of multiple times as we do now. (And that can matter, locally, in some cases with large numbers of small files).

Reply:
I think that'd be a good idea.